firefox-ios/Client/Frontend/Browser/BrowserViewController/State/BrowserViewControllerState.swift

```diff
@@ -68,7 +69,8 @@ struct BrowserViewControllerState: ScreenState, Equatable {
                   navigateTo: bvcState.navigateTo,
                   displayView: bvcState.displayView,
                   buttonTapped: bvcState.buttonTapped,
-                  microsurveyState: bvcState.microsurveyState)
+                  microsurveyState: bvcState.microsurveyState,
+                  showPasswordGenerator: bvcState.showPasswordGenerator)
```

Comment:
Should we maybe use `displayView` here instead of adding another variable?

Reply:
I like this as well, good call out @thatswinnie

Reply:
I considered doing this earlier -- the only issue I noticed is most of the logic that responds to state changes of `displayView` in `BrowserViewController` is behind the `isToolbarRefactorEnabled` feature flag (see attached image). I could definitely still do it with logic for handling the update of `displayType` for `passwordGenerator` specifically being segregated
<img width="782" alt="Screenshot 2024-09-30 at 1 05 23 PM" src="https://github.com/user-attachments/assets/2bc2b681-f23b-42fa-9a8d-0a04a15f680f">

Reply:
Can you refactor it so it's not toolbar specific? We should keep the actions that are toolbar specific behind the toolbar flag but that can be done in `handleDisplayActions` and `handleNavigationActions`. The method `executeToolbarActions` was already renamed to `executeNavigationAndDisplayActions` (on main) to allow for other actions to be handled here.